sdk/lib/rtl/priv.c

```diff
@@ -486,3 +486,78 @@ RtlAdjustPrivilege(IN ULONG Privilege,
 
     return STATUS_SUCCESS;
 }
+
+/*
+ * @implemented
+ */
+NTSTATUS
+NTAPI
+RtlRemovePrivileges(IN HANDLE TokenHandle,
+                    IN PULONG PrivilegesToKeep,
+                    IN ULONG PrivilegeCount)
+{
+    DPRINT("RtlRemovePrivileges(%p, %p, %u)\n", TokenHandle, PrivilegesToKeep, PrivilegeCount);
+
+    /* Save privileges that should be keep */
+    UINT64 PrivilegesToKeepBitmap = 0;
+    if (PrivilegeCount)
+    {
+        ULONG ulCount;
+        PULONG pulPrivilegesToKeep = PrivilegesToKeep;
+        for (ulCount = 0; ulCount < PrivilegeCount; ulCount++)
+        {
+            PrivilegesToKeepBitmap |= (1ULL << *pulPrivilegesToKeep);
+            if (*pulPrivilegesToKeep++ > SE_MAX_WELL_KNOWN_PRIVILEGE)
+            {
+                return STATUS_INVALID_PARAMETER;
+            }
+        }
+    }
+
+    /* Get token privileges information */
+    ULONG ulReturnLength;
+    BYTE tpBuffer[sizeof(TOKEN_PRIVILEGES) +
+                  sizeof(LUID_AND_ATTRIBUTES) * (SE_MAX_WELL_KNOWN_PRIVILEGE - SE_MIN_WELL_KNOWN_PRIVILEGE)];
+    NTSTATUS Status = ZwQueryInformationToken(TokenHandle,
+                                              TokenPrivileges,
+                                              tpBuffer,
+                                              sizeof(tpBuffer),
+                                              &ulReturnLength);
+    if (NT_SUCCESS(Status))
```

Comment:
Please do a return-early here instead, so that the (long) success code path doesn't get indented more.